### ISSUE TITLE
"module": "node12" becomes invalid with "moduleResolution": "node"

### ISSUE THREAD

**Issue by jeffrson** [microsoft/TypeScript#46202]
# Bug Report

### 🔎 Search Terms

node12
nodenext
moduleResolution

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions ______ and _______
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
- I was unable to test this on prior versions because it's a new feature since 4.5.0-beta

### 🙁 Actual behavior

Typescript does not follow "exports" from "package.json", when there's not only
```json
{
  "compilerOptions": {
    "module": "node12",  // or "nodenext"
    ...
  }
}

```
but also "moduleResolution": "node" in "compilerOptions".

Also, despite "type": "module" in package.json, import/export are not left alone in transpiled files, instead it looks like "module" is reset to "commonjs".

### 🙂 Expected behavior

If "moduleResolution" contradicts "module": "node12" there may be a warning or an error - or it could be simply ignored.

Since "moduleResolution" cannot be set to "node12" or "nodenext" so far, the only way to use the new resolution is to remove or comment it in order to make use of exports.

**Comment by simllll** [comment]
I ran into the same issue, documented it over here: https://github.com/microsoft/TypeScript/pull/44501#issuecomment-933669847

**Comment by andrewbranch** [comment]
> Since "moduleResolution" cannot be set to "node12" or "nodenext" so far

What? Why can't it?

**Comment by simllll** [comment]
As mentioned in #444501 `The new moduleResolution modes currently just redirect to the old node resolution mode until they're actually implemented in another PR.` I guess the new modes are actually not really new modes yet? But all I really know is that it's not possible to set moduleResolution to node12 right now ;-)

**Comment by jeffrson** [comment]
> > Since "moduleResolution" cannot be set to "node12" or "nodenext" so far
> 
> What? Why can't it?

ok - actually you can ;-) - but it results in an error message because it only accepts "node" and "classic" for now

**Comment by andrewbranch** [comment]
The new resolution modes are in the beta/nightly. The “error message” you’re referring to is just the JSON schema validation, and that ships separately from TypeScript, unfortunately. We will be updating the schema soon so it shows the new values as legal. But you can just ignore the message; TypeScript understands and accepts `node12` and `nodenext`.

**Comment by orta** [comment]
Yep, PR here: https://github.com/SchemaStore/schemastore/pull/1860

**Comment by andrewbranch** [comment]
Config parser is fixed in #46239—it should be in the nightly tonight. Good catch!
